[17.0][MIG] auditlog: Migration to 17.0

rven opened the pull request.

sebalix committed:
[ADD] Module 'auditlog' - A substitute to the deprecated 'audittrail' module

sebalix committed:
[FIX] Module 'auditlog' - Fix some flake8 errors

sebalix committed:
[FIX] Module 'auditlog' - Put the module description in a README.rst file

sebalix committed:
[ADD] Module 'auditlog' - 'auditlog.pot' file added

sebalix committed:
[IMP] Module 'auditlog' - Split the 'create_logs()' method in several ones to make overridding easier

sebalix committed:
[IMP] Module 'auditlog' - Clean up vim lines

sebalix committed:
[IMP] Module 'auditlog' - Add 'name' attributes on some filters and other minor fixes

sebalix committed:
[IMP] Module 'auditlog' - Removed the 'timestamp' field from the 'auditlog.log' model (standard 'create_date' field is used instead)

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (based on the OCA template)

sebalix committed:
[FIX] Module 'auditlog' - 'time' module imported but unused

hbrunn committed:
[ADD] init hook to migrate from audittrail

hbrunn committed:
[IMP] ignore rules for nonexisting objects

hbrunn committed:
[FIX] also migrate id sequences

sebalix committed:
[IMP] Module 'auditlog' - Contributors updated in README.rst + AUTHORS.txt file removed

hbrunn committed:
[ADD] change xmlids of legacy rules to point to auditlog.rule

hbrunn committed:
[IMP] use new import of _, SUPERUSER_ID

hbrunn committed:
[ADD] allow overriding modules to pass additional log values

hbrunn committed:
[IMP] cache model and field ids

hbrunn committed:
[IMP] don't make a temporary copy of dict keys

hbrunn committed:
[FIX] don't reset caches in register_hook

hbrunn committed:
[ADD] pass old_values to create_log when deleting a record

hbrunn committed:
[FIX] cope with no additional log values

hbrunn committed:
[ADD] tests

sebalix committed:
[FIX] Module 'auditlog' - local variable 'ir_model' is assigned to but never used

hbrunn committed:
[IMP] use a model for test that is not influenced by the mail module

sebalix committed:
[FIX] Module 'auditlog' - Remove 'active' field from __openerp__.py

sebalix committed:
[IMP] Module 'auditlog' - README.rst updated (inform that the audittrail data - if any - are migrated during the installation)

sebalix committed:
[IMP] Module 'auditlog' - Useless lines of code removed

sebalix committed:
[IMP] Module 'auditlog' - Autoremove 'auditlog.log.line' records when a field is deleted (e.g. migration)

sebalix committed:
[IMP] Module 'auditlog' - Model 'auditlog.log.line', 'field_name' and 'field_description' fields changed into related fields

sebalix committed:
[IMP] Module 'auditlog' - Added a second test using cached data generated during the first test

sebalix committed:
[FIX] Module 'auditlog' - Data dictionaries of fields and models are put in the cache instead of recordsets (a recordset contains a reference on the database cursor which is closed after each RPC query)

sebalix committed:
[IMP] Module 'auditlog' - Add a test regarding a resource deleted through a (2, ID) tuple on a write operation

sebalix committed:
[FIX] Module 'auditlog' - When updating a resource, filter IDs on 2many fields to prevent a 'name_get()' call on potential deleted resources

hbrunn committed:
[ADD] migration script

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

sebalix committed:
[IMP] Module 'auditlog' - Performing logs on 'read' operations + Some bugfixes to log inherited fields and dummy fields such as 'in_group_X' in 'res.users' model +  Unit tests updated

sebalix committed:
[FIX] Module 'auditlog' - Replace 'try/except statement by 'isinstance()' + Add missing field in unit test + Remove a list comprehension

sebalix committed:
[FIX] Module 'auditlog' - Support 'read' calls with one ID or a list of IDs

sebalix committed:
[FIX] Module 'auditlog' - Unit tests, do not create users with the 'notify_email' field

eqms committed:
[ADD] German translation for auditlog

sebalix committed:
[FIX] Module 'auditlog' - No log for internal processing (e.g. 'read' calls produced by auditlog, either to scan records data or to fetch informations from 'ir.model'/'ir.model.fields' data models if rules are defined on them)

sebalix committed:
[FIX] Module 'auditlog' - Fix lint check

sebalix committed:
[IMP] Module 'auditlog' - Unsubscribe rules before removing them + Unit tests updated

sebalix committed:
[FIX] Module 'auditlog' - Log fields coming from polymorphic inheritances ('_inherits')

sebalix committed:
[IMP] Module 'auditlog' - Remove the unit test causing trouble with Travis CI

Commit:
[ADD] auditlog: Added Spanish translation.

yvaucher committed:
Add bug tracker link on README.rst

hbrunn committed:
[FIX] ie related fields don't have an

ir.model.fields entry

hbrunn committed:
[ADD] comment

hbrunn committed:
[FIX] make patched create downgrade correctly to v6.1

yvaucher committed:
Add missing default oca icons

hbrunn committed:
[FIX] if someone else has overridden our patched,

method, don't do anything

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebalix committed:
Module 'auditlog' - Log HTTP user sessions and requests

sebalix committed:
Module 'auditlog' - Bump the version to 1.1.0

sebalix committed:
Module 'auditlog' - Cleanup XML views (types removed)

sebalix committed:
Module 'auditlog' - A log can be created with no current HTTP request (unit tests, ir.cron...)

sebalix committed:
Module 'auditlog' - 'display_name' fields added for the user session and HTTP requests models

sebalix committed:
Module 'auditlog' - Fix pylint check

hbrunn committed:
turn off prefetching when reding values

sebalix committed:
auditlog - Ability to choose the log type on the rule: Full log (complete but slow) and Fast log (data input only, faster)

sebalix committed:
auditlog - Field 'type' renamed to 'log_type'

sebalix committed:
auditlog - Auto-vacuum logs, HTTP requests and HTTP user sessions

sebalix committed:
auditlog - Bump version number

sebalix committed:
auditlog - Icon added + README updated (screenshots)

holdenrehg committed:
Migrate auditlog module from 8.0 to 9.0

- Update documentation to point to the new auditlog menu locations. These were changed because the 8.0 version was referencing menus that do not exist in 9.0
- Change version from 8.0.X.Y.Z to 9.0.1.0.0
- Make the module installable again
- Remove an unused parameter from pre-migration.py for versioning
- Fix typos and remove commented out blocks of code that were irrelevant

holdenrehg committed:
Updated the __openerp__.xml to include the license and images fields

These were updated to follow OCA conventions.

- License set to AGPL-3
- Images set to empty array

holdenrehg committed:
auditlog module, Remove duplicate license key from openerp config

holdenrehg committed:
auditlog: Remove migration from audittrail functionality, this is not required for version 9

holdenrehg committed:
auditlog: Remove pre_init_hook reference from openerp, no pre_init hook exists any more

hbrunn committed:
singleton error if we saved the current session two times (#473)

* [IMP] index the columns we'll be searching for for every request

* [FIX] singleton error if we saved the current session two times

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Rename manifest files

sebalix committed:
auditlog - Lint

sebalix committed:
auditlog - Migrate 'AuditlogRule.create' and 'AuditlogRule.write' methods to the new API

StefanRijnhart committed:
[FIX] Request id no longer exists after concurrency rollback

Commit:
[MIG] auditlog: Migrated to 10.0

obulkin committed:
[FIX] auditlog: Duplicate logging
* Update _patch_methods and _revert_methods in auditlog.rule model to properly
track whether a method has already been patched for logging purposes. This
prevents duplicate logs from being produced in various cases (e.g. when a
logging rule is created as part of a module install)

etobella committed:
[MIG] auditlog: Migration to 11.0

rgarnau committed:
[add] https

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update auditlog.pot

Commit:
Translated using Weblate

Currently translated at 100,0% (70 of 70 strings)

Translation: server-tools-11.0/server-tools-11.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-11-0/server-tools-11-0-auditlog/pt_BR/

sebalix committed:
[MIG] auditlog: Migration to 12.0

OCA-git-bot committed:
[UPD] README.rst

[UPD] Update auditlog.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-auditlog/

[UPD] README.rst

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (70 of 70 strings)

Translation: server-tools-12.0/server-tools-12.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-auditlog/zh_CN/

sebalix committed:
[FIX] auditlog: log computed fields stored in db as expected

Fixing #1134.
Odoo stores values of computed fields at the end of the transaction
only, as such performing a 'read()' to make a data snapshot on the record
created in the current transaction doesn't return the expected result
regarding these fields.
Also as a side-effect 'read()' alters the environment cache and break the
values on the record inducing issues in the whole user
transaction/workflow.

This fix replaces the use of 'read()' to do the data snapshot directly
from the cache of the record (computed values are already there).

lembregtse committed:
[MIG] auditlog: Migration to 13.0

sebalix committed:
[FIX] auditlog: add support for create multi

rven committed:
[IMP] auditlog: black, isort, prettier

oca-travis committed:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-auditlog/

emagdalenaC2i committed:
[FIX] Readme Images relative path

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auditlog 13.0.1.0.1

alan196 committed:
[IMP] auditlog: black, isort, prettier

alan196 committed:
[MIG] auditlog: Migration to 14.0

oca-travis committed:
[UPD] Update auditlog.pot

Commit:
[IMP] Dont query non-stored related or computed fields on full auditlog, this can slow down the system a lot and gives no useful info

OCA-git-bot committed:
auditlog 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

Commit:
[14.0][FIX] Add sudo() to retrieve record vals

OCA-git-bot committed:
auditlog 14.0.1.0.2

Commit:
[IMP] auditlog: prevent cascading delete of logs when models or fields are unlinked

When a field or a model is unlinked, keep the related audit logs. Denormalize
the field and model info on the logs and log lines so that the information
is still available after the deletion of the related data model.

Also, to improve the performance of the deletion of fields and models,
add indexes on the log's model_id and log line's field_id.

Co-Authored-By: Stefan Rijnhart <stefan@opener.am>

oca-travis committed:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-auditlog/

StefanRijnhart committed:
[FIX] auditlog: invalid XML

Fixes #2086

OCA-git-bot committed:
auditlog 14.0.1.1.1

Commit:
[ADD] auditlog: enable to track Unlink record

Commit:
[ADD] auditlog: test case for capture record

oca-travis committed:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-auditlog/

kittiu committed:
[14.0][IMP] auditlog, add auditlog.log.line view

oca-travis committed:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auditlog 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-auditlog/

Commit:
[IMP] auditlog: black, isort, prettier

Commit:
[MIG] auditlog: Migration to 15.0

Commit:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (81 of 81 strings)

Translation: server-tools-15.0/server-tools-15.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-auditlog/es_AR/

atchuthan committed:
Add option to Eliminate user and fields in audit logs

hbrunn committed:
[FIX] auditlog: Allow passing a chunk size for autovacuum

Commit:
[UPD] Update auditlog.pot

OCA-git-bot committed:
auditlog 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-15.0/server-tools-15.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-auditlog/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (83 of 83 strings)

Translation: server-tools-15.0/server-tools-15.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-auditlog/es_AR/

gurneyalex committed:
[FIX] auditlog: autovacuum performance

Add database indices on the foreign key fields of the auditlog models.
Without these indices, the performance of the autovacuum cron are terrible
because the "ON DELETE SET NULL" trigger has to make a full table scan
on the auditlog_log and auditlog_log_line tables when
auditlog_http_session and auditlog_http_request rows are deleted.

OCA-git-bot committed:
auditlog 15.0.1.1.1

StefanRijnhart committed:
[MIG] auditlog: Migration to 16.0

BertVGroenendael committed:
[IMP] auditlog: dedicated security groups for model access

SilvioC2C committed:
FIX: auditlog consistency with Many2one fields

Odoo allows creating or updating records with values like ``{"many2one_field_id": empty.recordset()}``,
but this crashes when using ``deepcopy``, which fails in recreating the ``api.Environment`` object
attached to the recordset.
This commit should fix the issue by updating the values before passing them to ``deepcopy``.

etobella committed:
[IMP] auditlog: make the line views using a non auto model

Commit:
[UPD] Update auditlog.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (86 of 86 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/es_AR/

Commit:
[UPD] Update auditlog.pot

OCA-git-bot committed:
auditlog 16.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (88 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/es_AR/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auditlog 16.0.2.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 36.3% (32 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 78.4% (69 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/sl/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 36.3% (32 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (88 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auditlog 16.0.2.0.2

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 36.3% (32 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/it/

rven committed:
[IMP] auditlog: pre-commit stuff

pedrobaeza commented:
/ocabot migration auditlog

rven commented:
A "big" remark here is that the methods `_patch_method` and `_revert_method` are not available anymore in Odoo.
I re-implemented them in the `base.py` file.
If somebody knows a better way, let me know!

pedrobaeza commented:
There should be then an alternative implementation. Have you dug into the commit removing it to check?

rven commented:
> There should be then an alternative implementation. Have you dug into the commit removing it to check?

Yes, they removed it, because it was only used in Odoo test classes.
Here they are using the patch method now from the test.common package, which is not the way to go I guess ;-)

pedrobaeza commented:
OK, I let others used to this module to check and give their opinion.

nilshamerlinck commented:
Hello @rven what about using the same approach as their fix on `base_automation` in [PR](https://github.com/odoo/odoo/pull/110370):

![image](https://github.com/OCA/server-tools/assets/511258/b50b96e1-9829-4a75-93cd-432954500437)

To avoid re-introducing these risky methods ;-)

rven commented:
@nilshamerlinck Thanks for the tip!

ahaly-compass commented:
Whats the status here?

esakarya commented:
Not working properly, when u try open Logs, Menü happen following error. also delete never logged,

 File "/opt/odoo/odoo-enterprise/odoo/models.py", line 2717, in read_group
    rows = self._read_group(domain, annoted_groupby.values(), annoted_aggregates.values(), offset=offset, limit=limit, order=orderby)
  File "/opt/odoo/odoo-enterprise/odoo/models.py", line 1906, in _read_group
    self.env.cr.execute(SQL("\n").join(query_parts))
  File "/opt/odoo/odoo-enterprise/odoo/sql_db.py", line 332, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.SyntaxError: syntax error at or near "SELECT"
LINE 2: FROM SELECT 
             ^

rven commented:
@esakarya The error should be fixed now.

Are you sure the unlink of a record is not logged?
![image](https://github.com/OCA/server-tools/assets/8638295/484b91ee-33cd-464f-bab7-ccfb37692491)

StefanRijnhart committed:
[FIX] auditlog: prevent removal of x2many values from inaccessible companies

Having a product.template with taxes from multiple companies assigned, let a
user without all of these companies enabled set the taxes of that product.

Before this change, the taxes related to inaccessible companies will be
inadvertently removed.

Co-authored-by: Raf Ven <raf.ven@dynapps.be>

StefanRijnhart reviewed (approved):
Looking good, and seems to work fine on runboat.

StefanRijnhart commented on auditlog/models/autovacuum.py:
```diff
@@ -0,0 +1,37 @@
+                order="create_date asc",
+            )
+            nb_records = len(records)
+            with self.env.norecompute():
```
```
2024-02-02 12:22:58,062 3016808 WARNING 17auditlog py.warnings: /usr/lib/python3.10/contextlib.py:135: DeprecationWarning: `norecompute` is useless. Deprecated since 17.0.
  File "/home/odoo/server-tools/auditlog/tests/test_autovacuum.py", line 40, in test_autovacuum
    autovacuum_model.autovacuum(days=0)
  File "/home/odoo/server-tools/auditlog/models/autovacuum.py", line 34, in autovacuum
    with self.env.norecompute():
  File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
```

rven replied:
Removed!

StefanRijnhart reviewed (commented):
Oh, tests emit a deprecation warning

StefanRijnhart commented:
If you want you can pick https://github.com/OCA/server-tools/pull/2814 here. It's a clean pick onto your branch and the issue still applies.

rven committed:
[MIG] auditlog: Migration to 17.0

StefanRijnhart reviewed (approved):
Thanks for the updates!

BT-anieto committed:
[16.0][FIX] auditlog: Add/Remove User Group

When adding/removing a group in a user (and tracking a user or a partner), odoo sends a 'reified' val 'in_group_{group_id}'. We must convert this value into the real groups_id one.

xFingerlinKx commented:
when will the version for 17 Odoo be available?

StefanRijnhart commented:
@xFingerlinKx as soon as more people have done a code review or a functional test. People like you!

StefanRijnhart commented:
@rven please check this fix of a regression introduced by the cache invalidation improvement: https://github.com/OCA/server-tools/pull/2834

rven commented:
@StefanRijnhart I added the specific commit into this PR, I had to change the test a little bit because of a ensure_one error.

StefanRijnhart commented:
@rven Great, thanks!

heccjj commented:
Is this a bug?
I override the method do_print_picking() of stock.picking model, and there is code like this:
self.write({'printed': True})
Then add and subscribe stock.picking auditlog rule.
When I print a stock picking, I will get an empty log, without log_line!

StefanRijnhart commented:
@heccjj Your change does not seem substantial, you just leave out actually rendering the report. Therefore, I tested this on this PR's runbot, and I can't reproduce:

![image](https://github.com/OCA/server-tools/assets/1033124/3b5343d2-0a13-40a7-af41-03dafd1ddcc2)

So it does not seem like a bug in the module. Perhaps there is some other customization in your setup that interferes here.

heccjj commented:
@StefanRijnhart  Thanks for your reply. I try it again. If I print the picking slip first time, I get correct log, but if I print this picking slip again(field printed is already "True"), get an empty log.

StefanRijnhart commented:
@heccjj That makes sense, because the written values are only logged if the values differ from the current field values. The log is still present to represent the write itself (this is a design choice), but it is empty because there were no actual changes. So, not a bug.

DucTruongKomit reviewed (approved):
Look good

pedrobaeza commented:
/ocabot merge nobump
/ocabot migration auditlog
